www/src/views/XMultiCredentialView.tsx

```diff
@@ -0,0 +1,151 @@
+      kind
+      fails
+      target {
+        id
+        name
+      }
+    }
+  }
+`;
+
+const REMOVE_CREDENTIAL = gql`
+  mutation RemoveCredential($credential: ID!) {
+    deleteCredential(input: { id: $credential })
+  }
+`;
+
+type RemoveCredentialButtonProps = {
+  credential: Credential;
+};
+
+const RemoveCredentialButton: FunctionComponent<RemoveCredentialButtonProps> = ({
+  credential
+}) => {
+  const [removeCredential] = useMutation(REMOVE_CREDENTIAL, {
+    refetchQueries: [{ query: MULTI_CREDENTIAL_QUERY }]
+  });
+
+  const handleRemove = () => {
+    let vars = {
+      credential: credential.id
+    };
+
+    removeCredential({
+      variables: vars
+    });
+  };
+
+  return (
+    <Button.Group labeled>
+      <Button compact negative icon="x" onClick={handleRemove} />
+    </Button.Group>
+  );
+};
+
+export type MultiCredentialResponse = {
+  credentials: Credential[];
+};
+
+type XCredentialTableRowProps = {
+  credential: Credential;
+};
+
+const XCredentialTableRow: FunctionComponent<XCredentialTableRowProps> = ({
+  credential
+}) => {
+  const whenEmpty = <span>No Credentials</span>;
+
+  return (
+    <Table.Row>
+      <Table.Cell width={2}>
+        <b>{credential.principal}</b>
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        <b>{credential.secret}</b>
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        {credential.kind}
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        {credential.target.name}
+      </Table.Cell>
+      <Table.Cell collapsing width={8}>
+        <RemoveCredentialButton credential={credential} />
+      </Table.Cell>
+    </Table.Row>
+  );
+};
+
+const XMultiCredentialView = () => {
+  const { loading, error, data: { credentials = [] } = {} } = useQuery<
+    MultiCredentialResponse
+  >(MULTI_CREDENTIAL_QUERY, {
+    pollInterval: 5000
+  });
+
+  const whenLoading = (
+    <XLoadingMessage
+      title="Loading Credentials"
+      msg="Fetching Credential info"
+    />
+  );
+  const whenEmpty = (
+    <Table.Row>
+      <Table.HeaderCell colSpan="5">
+        <XNoCredentialsFound />
+      </Table.HeaderCell>
+    </Table.Row>
+  );
+
+  return (
+    <React.Fragment>
+      <XErrorMessage title="Error Loading Credentials" err={error} />
+
+      <XBoundary boundary={whenLoading} show={!loading}>
+        <Table celled style={{ overflow: "auto" }}>
+          <Table.Header>
+            <Table.Row>
+              <Table.HeaderCell>Credential</Table.HeaderCell>
+              <Table.HeaderCell>Secret</Table.HeaderCell>
+              <Table.HeaderCell>Type</Table.HeaderCell>
+              <Table.HeaderCell>Target</Table.HeaderCell>
+              <Table.HeaderCell></Table.HeaderCell>
+            </Table.Row>
+          </Table.Header>
+
+          <Table.Body>
+            <XBoundary
+              boundary={whenEmpty}
+              show={credentials && credentials.length > 0}
+            >
+              {credentials &&
```

Comment:
It would be dope if we could do some aggregation, i.e. only show each principal secret pair once and include a list of targets.